python/pyiceberg/avro/decoder.py

```diff
@@ -50,10 +50,10 @@ def read(self, n: int) -> bytes:
             raise ValueError(f"Requested {n} bytes to read, expected positive integer.")
         read_bytes = self._input_stream.read(n)
         read_len = len(read_bytes)
-        if read_len <= 0:
-            raise EOFError
+        if read_len < 0:
+            raise EOFError(f"Got negative length: {read_len}")
         elif read_len != n:
-            raise ValueError(f"Read {len(read_bytes)} bytes, expected {n} bytes")
+            raise EOFError(f"Read {read_len} bytes, expected {n} bytes")
```

Comment:
I don't think this is quite correct. If read_len is 0 then `EOFError` is correct. But it is perfectly fine for a read to return fewer bytes than requested. From [RawIOBase.read docs](https://docs.python.org/3/library/io.html#io.RawIOBase.read):

> Read up to size bytes from the object and return them. As a convenience, if size is unspecified or -1, all bytes until EOF are returned. Otherwise, only one system call is ever made. Fewer than size bytes may be returned if the operating system call returns fewer than size bytes.

If fewer bytes are returned this can either loop and call read until the required bytes are returned ("read fully"), or it can return a shorter array. Since this is a higher-level API, I think it should have read fully behavior.